runtime/js/marshal.js

```diff
@@ -880,10 +880,10 @@ function caml_output_value_to_bytes(v, flags) {
 }
 
 //Provides: caml_output_value_to_buffer
-//Requires: caml_output_val, caml_failwith, caml_blit_bytes
+//Requires: caml_output_val, caml_failwith, caml_blit_bytes, caml_bytes_of_array
 function caml_output_value_to_buffer(s, ofs, len, v, flags) {
   var t = caml_output_val(v, flags);
   if (t.length > len) caml_failwith("Marshal.to_buffer: buffer overflow");
-  caml_blit_bytes(t, 0, s, ofs, t.length);
```

Comment:
Can you confirm this is a bugfix ? Can we have a regression tests ?

Reply:
Indeed.